translation-service/src/main/java/eu/europeana/api/translation/service/GoogleTranslationService.java

```diff
@@ -0,0 +1,164 @@
+        this.client = TranslationServiceClient.create(translationServiceSettings);
+      } else {
+        TransportChannelProvider transportChannelProvider = CloudTasksStubSettings
+            .defaultGrpcTransportProviderBuilder()
+            .setChannelConfigurator(
+                managedChannelBuilder -> managedChannelBuilder.proxyDetector(socketAddress -> null))
+            .build();
+
+        TranslationServiceSettings tss;
+        tss = TranslationServiceSettings.newBuilder()
+            .setTransportChannelProvider(transportChannelProvider).build();
+
+        this.client = TranslationServiceClient.create(tss);
+      }
+
+      this.locationName = LocationName.of(getGoogleProjectId(), "global");
+      LOG.info("GoogleTranslationService initialised, projectId = {}", getGoogleProjectId());
+    } catch (IOException e) {
+      throw new RuntimeException("Cannot instantiate Google TranslationServiceClient!", e);
+    }
+  }
+  
+  /**
+   * used mainly for testing purposes. 
+   * @param client
+   */
+  public void init(TranslationServiceClient client) {
+    this.client = client;
+    this.locationName = LocationName.of(getGoogleProjectId(), "global");
+  }
+
+  // @PreDestroy
+  @Override
+  public void close() {
+    if (this.client != null) {
+      LOG.debug("Shutting down GoogleTranslationService client...");
+      this.client.close();
+    }
+  }
+
+  // public List<String> translate(List<String> texts, String targetLanguage,
+  // Language sourceLangHint) {
+  // TranslateTextRequest request = TranslateTextRequest.newBuilder()
+  // .setParent(locationName.toString()).setMimeType(MIME_TYPE_TEXT)
+  // .setTargetLanguageCode(targetLanguage).addAllContents(texts).build();
+  // TranslateTextResponse response = this.client.translateText(request);
+  // List<String> result = new ArrayList<>();
+  // for (Translation t : response.getTranslationsList()) {
+  // result.add(t.getTranslatedText());
+  // }
+  // return result;
+  // }
+
+  @Override
+  public List<String> translate(List<String> texts, String targetLanguage)
+      throws TranslationException {
+    return translate(texts, targetLanguage, null);
+  }
+
+  @Override
+  public List<String> translate(List<String> text, String targetLanguage, String sourceLanguage) {
+    Builder requestBuilder = TranslateTextRequest.newBuilder().setParent(locationName.toString())
+        .setMimeType(MIME_TYPE_TEXT).setTargetLanguageCode(targetLanguage).addAllContents(text);
+
+    if (sourceLanguage != null) {
+      requestBuilder.setSourceLanguageCode(sourceLanguage);
+    }
+
+    TranslateTextRequest request = requestBuilder.build();
+
+    TranslateTextResponse response = this.client.translateText(request);
+    List<String> result = new ArrayList<>();
+    for (Translation t : response.getTranslationsList()) {
+      result.add(t.getTranslatedText());
+    }
+    return result;
+  }
+
+  @Override
+  public boolean isSupported(String srcLang, String targetLanguage) {
```

Comment:
this method always returns true. Is it intended like that as we yet don't have logic to validate the languages supported?

Reply:
no specs available yet, will need to check with Hugo